### ISSUE TITLE
Stats & Rcon ports not working via docker-compose

### ISSUE THREAD

**Issue by saurabh21289** [quakelive-server-standards/quakelive-server-standards#23]
After running "docker-compose up" the server is up and running and everything works as expected apart from the stats and rcon ports. Do the bind failed errors in the log below mean anything?

```
myservers_ca1_1 | Hostname: 9fe1d28bb1b8
myservers_ca1_1 | IP: 172.18.0.3
myservers_ca1_1 | I: 22-07-21 23:32:16 zauth: API command=PLAIN
myservers_ca1_1 | [minqlx.late_init] INFO: Loading preset plugins...
myservers_ca1_1 | [minqlx.load_plugin] INFO: Loading plugin 'balance'...
myservers_ca1_1 | [minqlx.load_plugin] INFO: Loading plugin 'docs'...
myservers_ca1_1 | [minqlx.load_plugin] INFO: Loading plugin 'essentials'...
myservers_ca1_1 | [minqlx.load_plugin] INFO: Loading plugin 'log'...
myservers_ca1_1 | [minqlx.load_plugin] INFO: Loading plugin 'permission'...
myservers_ca1_1 | [minqlx.load_plugin] INFO: Loading plugin 'plugin_manager'...
myservers_ca1_1 | [minqlx.load_plugin] INFO: Loading plugin 'commands'...
myservers_ca1_1 | [minqlx.load_plugin] INFO: Loading plugin 'listmaps'...
myservers_ca1_1 | zmq RCON socket error, bind failed: tcp://:28960
myservers_ca1_1 | stdin is not a tty, tty console mode failed
myservers_ca1_1 | ------ Server Initialization ------
myservers_ca1_1 | Server: evolution
myservers_ca1_1 | zmq PUB socket error, bind failed: tcp://107.34.58.34:27960
myservers_ca1_1 | Hunk_Clear: reset the hunk ok
myservers_ca1_1 | ----- FS_Startup -----
myservers_ca1_1 | Current search path:
myservers_ca1_1 | /home/steam/.quakelive/baseq3
myservers_ca1_1 | /home/steam/Steam/steamapps/common/Quake Live Dedicated Server/baseq3/pak00.pk3 (0x15a37833 - 9285 files)
myservers_ca1_1 | /home/steam/Steam/steamapps/common/Quake Live Dedicated Server/baseq3/bin.pk3 (0x7c1d0d5e - 5 files)
myservers_ca1_1 | /home/steam/Steam/steamapps/common/Quake Live Dedicated Server/baseq3
myservers_ca1_1 | ./qzeroded.x64/baseq3
```
Here's my docker-compose for reference. I followed the steps in the ReadMe.md
[docker-compose-stats-rcon-not-working.yml.txt](https://github.com/quakelive-server-standards/quakelive-server-standards/files/9170192/docker-compose-stats-rcon-not-working.yml.txt)

**Comment by garzfaust** [comment]
Hi Saurabh,

thank you for posting the problem. I downloaded your Docker Compose file and tried to recreate the error. While doing so, i found various other problems which i fixed.

1. The Docker service for Clan Arena defined in the `docker-compose.source.yml` file is missing the `volumes` statement. This means, that you can remove the corresponding lines in your Docker Compose file which are still in the `volumes` section. Fixed it here: 35f32070b1e5a23d4185750b11dcd24e0277adf9
2. The standard `.factories` file for Clan Arena was missing a `,` and had one too much. Fixed it here: ec7e6b494ab27690b99c587835929dfac09949fc
3. The automatic plugin detection did not work. Fixed it here: b0ed59e5fb9c022c6a063cf3d159e0c802313515
4. The definition of empty environment variables in the `environment` section of the Docker Compose file like `- ZMQ_STATS_PORT=` disables the definition on the next line like which is happening in this case with the `SV_HOSTNAME` definition. Fixed it here: 80ff240565e415fda430d767e51de280825e59ab
5. To successfully use a `#` symbol for the host name, the whole line with the environment variable definition needs to be wrapped in `'`, not just the value. Fixed it here: e0715d925f65411d9227aa97102b33ccbb8659bf
6. Added support for the `SERVERSTARTUP` environment variable as i could see that you wanted to use it. Added it here: a82082c57c10bbd1535d6e62d802f1c11310de73

As for your problem of not working rcon and stats ports. After the above fixes, the stats API started working again. The rcon API remained not working though. The  problem was that `zmq_rcon_ip` had a value of `""` which the Quake Live dedicated server does not seem to interpret as `no value` which should trigger the default value. Instead it uses an empty value which does not work. Now the values of all IP's are `0.0.0.0`. Fixed it here: 43862c3cef78d330e9c38810f9ab0c3a78951808

If there are any more problems, let me know.

**Comment by saurabh21289** [comment]
Thanks @garzfaust , the problems I ran into are now resolved. Thank you for the fixes. I am now seeing only one issue which is with the ql-console,

`
~/git/quakelive-server-standards/_myservers$ sudo ./connect.sh <masked> --rcon-port 28960 --rcon-password <masked> --stats-port 27960 --stats-password <masked>
docker: Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "/ql-console/connect.sh": stat /ql-console/connect.sh: no such file or directory: unknown.
`
